TBD_Main_Branch/src/main/java/ics372/controllers/MainController.java

```diff
@@ -27,11 +35,52 @@ public MainController(DataService dataService, GsonService gsonService){
 
     public List<Warehouse>  getWarehouseList(){ return warehouseList;}
 
-    public String processJsonInputFile(String file){
+    public String processInputFile(String file){
         Collection<Shipment> shipmentList = new ArrayList<>();
         String msg = "";
 
-        shipmentList.addAll(gson.processJsonInputFile(file));
+        File f = new File(file);
+        String ext = FilenameUtils.getExtension(f.getName());
+
+        // checks if the file has the xml ending, if it does parse it, otherwise assume its a json file
+        if(ext.equals("xml")) {
+            try {
+                // turns the xml into usable data docs
+                DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+                Document doc = dBuilder.parse(f);
+                doc.getDocumentElement().normalize();
+
+                // Gets all the shipment lists and puts them into a nodelist
+                NodeList shipmentNodeList = doc.getElementsByTagName("Shipment");
+
+                // Loop through each shipment in the nodelist
+                for (int index = 0; index < shipmentNodeList.getLength(); index++) {
+                    Node shipment = shipmentNodeList.item(index);
+
+                    if (shipment.getNodeType() == Node.ELEMENT_NODE) {
+                        Element eElement = (Element) shipment;
+
+                        // Gets all the values needed
+                        String warehouseID = ((Element) (shipment.getParentNode())).getAttribute("id"); // checks the shipment node parent, aka warehouse, and finds its ID
+                        String shipmentID =  eElement.getAttribute("id"); // grabs the id attribute
+                        String shipmentMethod = eElement.getAttribute("type"); // grabs the type attribute, aka air, rail, truck, etc.
+                        Double weight = Double.parseDouble(doc.getElementsByTagName("Weight").item(index).getTextContent()); // gets the weight by index
+                        Long receiptDate = Long.parseLong(doc.getElementsByTagName("ReceiptDate").item(index).getTextContent()); // gets the receipt date by index
```

Comment:
There's an issue with the code at the moment. The XML the prof gave us has an error where its missing a receipt date tag. This line of code gets the index of each receipt date tag. But the file is missing one, so it goes to the first tag it sees. There's four total shipments in one warehouse, but only three receipt date tags. So, once it loops the fourth time, it'll look for the fourth tag that doesn't exist, getting a null error. I haven't figure out how to solve this bug, any help would be great. Once this is solved, it should all work perfectly.